### PR TITLE
Move documentation from README to ReadTheDocs

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
Documentation was difficult to read in README

### How was it fixed?
Moved docs to ReadTheDocs/Sphinx
Added doctest to the utilities examples

#### Cute Animal Picture
image

### CODE REVIEW

**Comment by njgheorghita** [comment]
@pipermerriam I think I need write access to setup the webhook for ReadTheDocs

**Comment by njgheorghita** [comment]
@pipermerriam added doctest to the examples, this should be ready to merge if it gets a 👍

**Comment by pipermerriam** [comment]
Oh SNAP, those doctests are sexy.  ~250 code examples tested ftw.

File: docs/utilities.rst [hunk 1]
```diff
@@ -0,0 +1,1421 @@
...
+
+Returns the 32 byte log topic for the given event abi.
+
+.. doctest::
+
+   >>> from eth_utils import event_abi_to_log_topic
+   >>> event_abi_to_log_topic({'type': 'event', 'anonymous': False, 'name': 'MyEvent', 'inputs': []})
+   b'M\xbf\xb6\x8bC\xdd\xdf\xa1+Q\xeb\xe9\x9a\xb8\xfd\xedb\x0f\x9a\n\xc21B\x87\x9aO\x19*\x1byR\xd2'
+
+``event_signature_to_log_topic(event_signature)`` -> bytes
```
Comment (pipermerriam) [inline docs/utilities.rst]: Not for this PR but we should look at migrating all of these into docstrings in the functions themselves and use `autodoc` to populate the docs.  Probably easier to maintain that way and our code should be easier to read as well.  I suspect the doctests could even be embedded.

Comment (carver) [reply docs/utilities.rst]: TIL about the pytest args for running code examples in docstrings:
https://github.com/ethereum/pyrlp/pull/97/files#diff-513c9b6861fc981fb2c708db477f542cL2

File: tox.ini [hunk 2]
```diff
@@ -11,10 +12,14 @@ exclude= tests,venv
 usedevelop=True
 commands=
     py.test {posargs:tests}
+	doctest: make -C {toxinidir}/docs doctest
```
Comment (carver) [inline tox.ini]: Tabs? :grimacing:

File: docs/utilities.rst (hunk 1 above)
Comment (njgheorghita) [reply docs/utilities.rst]: Ahh cool, good to know thanks!!

**Pull request merged**